internal/services/mssql/mssql_managed_instance_security_alert_policy_resource.go

```diff
@@ -0,0 +1,378 @@
+package mssql
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/v5.0/sql"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceMsSqlManagedInstanceSecurityAlertPolicy() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceMsSqlManagedInstanceSecurityAlertPolicyCreate,
+		Read:   resourceMsSqlManagedInstanceSecurityAlertPolicyRead,
+		Update: resourceMsSqlManagedInstanceSecurityAlertPolicyUpdate,
+		Delete: resourceMsSqlManagedInstanceSecurityAlertPolicyDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ManagedInstancesSecurityAlertPolicyID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"managed_instance_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ValidateMsSqlServerName,
+			},
+
+			"disabled_alerts": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Set:      pluginsdk.HashString,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"Sql_Injection",
+						"Sql_Injection_Vulnerability",
+						"Access_Anomaly",
+						"Data_Exfiltration",
+						"Unsafe_Action",
+						"Brute_Force",
+					}, false),
+				},
+			},
+
+			"email_account_admins": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"email_addresses": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+				},
+				Set: pluginsdk.HashString,
+			},
+
+			"retention_days": {
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				Default:      0,
+				ValidateFunc: validation.IntAtLeast(0),
+			},
+
+			"state": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.SecurityAlertPolicyStateDisabled),
+					string(sql.SecurityAlertPolicyStateEnabled),
+					string(sql.SecurityAlertPolicyStateNew),
```

Comment:
this doesn't seem like something a user would set? what happens if it is passed in?

Reply:
@katbyte thanks for your feedback. Fixed.